src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java

```diff
@@ -1072,6 +1074,12 @@ public ResultSet getColumns(String c, String s, String tblNamePattern, String co
                             }
                         }
 
+                        // remove (optional) length/ dimension of the column before returning result set
+                        Matcher regexMatcher = TYPE_WITH_SIZE.matcher(colType);
```

Comment:
can we use a more contextual variable name here?

Reply:
Please see updated code, which does not use regular expressions.